build/util/src/lib.rs

```diff
@@ -85,6 +90,47 @@ pub fn target_board() -> Option<String> {
 /// Exposes the board type from the `HUBRIS_BOARD` envvar into
 /// `cfg(target_board="...")`.
 pub fn expose_target_board() {
+    let mut boards = vec![];
+    let mut out_dir =
+        std::path::PathBuf::from(std::env::var("OUT_DIR").unwrap());
+    loop {
+        let done = out_dir.file_name() == Some(OsStr::new("target"));
+        out_dir.pop();
+        if done {
+            break;
+        }
+    }
+    out_dir.push("boards");
+    println!("cargo:rerun-if-changed={}", out_dir.display());
+    if let Ok(dir) = std::fs::read_dir(&out_dir) {
+        for dirent in dir {
+            let Ok(dirent) = dirent else {
+                continue;
+            };
+            let path = dirent.path();
+
+            if path.extension() != Some(OsStr::new("toml")) {
+                // Ignore other files, such as READMEs, in the directory.
+                continue;
+            }
+
+            if let Some(stem) = path.file_stem() {
+                let stem = stem.to_string_lossy();
+                boards.push(format!("\"{stem}\""));
+            }
+        }
+    } else {
+        eprintln!(
+            "warning: boards directory not found, \
+                  target_board can't validate"
+        );
+    }
+    boards.sort();
+    boards.dedup();
```

Comment:
**nitpick:** Duplicates would probably be a mistake of some sort, maybe worth a warning print as well?

Reply:
Wait, is it even possible to get duplicates since we take only toml files' stems and don't do eg. lowercasing? Shouldn't all the file stems be necessarily unique?